gobblin-config-management/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,273 @@
+package gobblin.config.client;
+
+import gobblin.config.configstore.ConfigStore;
+import gobblin.config.configstore.ConfigStoreFactory;
+import gobblin.config.configstore.ConfigStoreWithImportedBy;
+import gobblin.config.configstore.ConfigStoreWithImportedByRecursively;
+import gobblin.config.configstore.ConfigStoreWithResolution;
+import gobblin.config.configstore.ConfigStoreWithStableVersion;
+import gobblin.config.configstore.impl.SimpleConfigStoreResolver;
+import gobblin.config.configstore.impl.SimpleImportMappings;
+import gobblin.config.utils.PathUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.TreeMap;
+
+import org.apache.log4j.Logger;
+
+import com.typesafe.config.Config;
+
+
+/**
+ * ConfigClient is the consumer facing class for retrieving configuration against all configuration stores.
+ * @author mitu
+ *
+ */
+@SuppressWarnings("rawtypes")
+public class ConfigClient {
+
+  private static final Logger LOG = Logger.getLogger(ConfigClient.class);
+
+  public static enum VERSION_STABILITY_POLICY {
```

Comment:
This should be named VersionStabilityPolicy. We generally reserve ALL_CAPS for constants.